File: src/engine/scoring.ts
import type { PlayerState } from '../data/types';

export function calculateScore(player: PlayerState): number {
  return player.completedGarments.reduce((sum, g) => sum + g.card.stars, 0) + player.ducats;
}

export function calculateScores(players: PlayerState[]): { name: string; score: number }[] {
  return players.map(p => ({ name: p.name, score: calculateScore(p) }));
}

export function determineWinner(players: PlayerState[]): string {
  const scores = calculateScores(players);
  scores.sort((a, b) => b.score - a.score);
  return scores[0].name;
}

File: colori-ai-wasm/src/scoring.rs
use crate::types::{AnyCard, PlayerState};

pub fn calculate_score(player: &PlayerState) -> u32 {
    let buyer_stars: u32 = player
        .completed_buyers
        .iter()
        .map(|ci| match &ci.card {
            AnyCard::Buyer { stars, .. } => *stars,
            _ => 0,
        })
        .sum();
    buyer_stars + player.ducats
}
